src/main/java/org/example/servletsHomework/servlet/RegisterServlet.java

```diff
@@ -0,0 +1,44 @@
+package org.example.servletsHomework.servlet;
+
+
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.example.servletsHomework.model.User;
+import org.example.servletsHomework.storage.TokensAndUserStorage;
+
+import java.io.IOException;
+
+public class RegisterServlet extends HttpServlet {
+
+    private final TokensAndUserStorage tokensAndUserStorage;
+
+    public RegisterServlet(TokensAndUserStorage tokensAndUserStorage) {
+        this.tokensAndUserStorage = tokensAndUserStorage;
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String username = request.getHeader("username");
+        String password = request.getHeader("password");
+
+        if (tokensAndUserStorage.ifUserExists(username)) {
+            response.setStatus(HttpServletResponse.SC_CONFLICT);
+            response.getWriter().write("RegisterServlet2: username already exists. Need log in");
+
+        } else {
+            User newUser = tokensAndUserStorage.createNewUser(username, password);
+            String newToken = tokensAndUserStorage.generateToken();
+            newUser.setCreationTime(System.currentTimeMillis());
```

Comment:
why can't it be inside `tokensAndUserStorage.createNewUser` ? or it is logic that create user without creation date, then create token and as 3rd step set creation date to user ?  You sure it is right sequence of actions ?